src/components/events/EventModal/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { Typography } from '@/components/common';
+import CommunityLogo from '@/components/common/CommunityLogo';
+import CalendarButtons from '@/components/events/CalendarButtons';
+import PointsDisplay from '@/components/events/PointsDisplay';
+import { PublicEvent } from '@/lib/types/apiResponses';
+import { formatEventDate } from '@/lib/utils';
+import LinkIcon from '@/public/assets/icons/link.svg';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useEffect, useRef } from 'react';
+import styles from './style.module.scss';
+
+interface EventModalProps {
+  open: boolean;
+  attended: boolean;
+  event: PublicEvent;
+  onClose: () => void;
+}
+
+const EventModal = ({ open, attended, event, onClose }: EventModalProps) => {
+  const { cover, title, start, end, location, description, eventLink } = event;
+
+  let displayCover = cover;
+  if (!displayCover) {
+    displayCover = '/assets/graphics/store/hero-photo.jpg';
+  }
+
+  let displayEventLink = eventLink;
+  if (!eventLink) {
+    displayEventLink = `https://acmucsd.com/events/${event.uuid}`;
+  }
```

Comment:
```suggestion
  let displayCover = cover || '/assets/graphics/store/hero-photo.jpg';

  let displayEventLink = eventLink || `https://acmucsd.com/events/${event.uuid}`;
```

you could also use `??` instead of `||` if you're feeling fancy

Reply:
Should this also use `formatURLEventTitle` to include the event title in the URL, like what is used for the Add to Apple Calendar button?

Reply:
hm `displayEventLink` just links to a main website preview of the event, so we need to keep it this way by convention, so probably not